### PR TITLE
Fix/large images android

### PR DESCRIPTION
**Pull request by ArekChr**
### Fixed Issues

$ #12893
PROPOSAL: https://github.com/Expensify/App/issues/12893#issuecomment-1446422809

### Tests
1. Go to any chat and upload a very long/large image. [Example image](https://user-images.githubusercontent.com/44479856/132889916-e46ac3dc-19ad-4cca-a033-ec13b8ac79ad.jpg)
2. In the Attachment preview, tap the image to zoom
3. Keep zooming until you'll see that the image is blurry

- [x] Verify that no errors appear in the JS console

### Offline tests
No offline tests needed

### QA Steps
1. Go to any chat and upload a very long/large image. [Example image](https://user-images.githubusercontent.com/44479856/132889916-e46ac3dc-19ad-4cca-a033-ec13b8ac79ad.jpg)
2. In the Attachment preview, tap the image to zoom
3. Keep zooming until you'll see that the image is blurry

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/24796318/227942965-72585383-2c48-4778-b505-105cbfd507cf.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/24796318/227942937-f96232a3-0f43-43a8-9bfe-e3073c6af8ca.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/24796318/227942702-7b335378-1e2d-47af-989c-50afc8e93e4c.mov

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/24796318/227942926-d0158864-cb49-4cce-99e0-01625ceca53d.mov

</details>

### CODE REVIEW

**Comment by ArekChr** [comment]
I found a bug; zoom-in doesn't work on mobile web chrome(android) and safari(iOS); zoom-in doesn't work.
I can't test these platforms

**Comment by cristipaval** [comment]
Hey @mananjadhav ! Could you please review this PR? I added you because you are assigned to the [issue](https://github.com/Expensify/App/issues/12893).

**Review by cristipaval — DISMISSED** [review]

**Comment by cristipaval** [comment]
FYI @mananjadhav , there is a known [bug](https://github.com/Expensify/App/issues/2499) which blocks you to test the zoom-in feature on mobile web, in case you want to test on all platforms, although the current PR is meant to fix a bug which occurs on Android native only.

**Comment by francoisl** [comment]
I was reading [this Slack thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1678447109985879?thread_ts=1670596654.583519&cid=C01GTK53T8Q) to get more context and it mentions creating a PR in the upstream repo, did we do that already?

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -10,6 +10,8 @@
 
     <application
       android:supportsRtl="false"
+      android:largeHeap="true"
+      android:hardwareAccelerated="false"
```
Comment (francoisl) [inline android/app/src/main/AndroidManifest.xml:14]: Just to make sure I understand correctly, why are we disabling hardware acceleration? The proposal [here](https://github.com/Expensify/App/issues/12893#issuecomment-1446422809) says:

> Displaying images larger than 8k pixels will also require enabling hardware acceleration and a large heap.

Comment (ArekChr) [reply android/app/src/main/AndroidManifest.xml:14]: @francoisl I made a mistake in the proposal. With enabled `hardwareAccelerated` together with `largeHeap` app won't display larger images over 8k pixels. 

Here are details of the native error `java.lang.RuntimeException: Canvas: trying to draw too large(394141296bytes) bitmap.`

**Comment by ArekChr** [comment]
> I was reading [this Slack thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1678447109985879?thread_ts=1670596654.583519&cid=C01GTK53T8Q) to get more context and it mentions creating a PR in the upstream repo, did we do that already?

Yes, the handling large images feature has been merged to Expensify/react-native-fast-image

File: android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (francoisl) [reply android/app/src/main/AndroidManifest.xml:14]: Got it, thanks for clarifying.

**Comment by francoisl** [comment]
@mananjadhav any chance you can help with the testing here please?

**Comment by francoisl** [comment]
Sorry for the delay here. There's a conflict in package-lock.json, can you update the PR please @ArekChr?


@mananjadhav any chance you can help with the testing here please?

**Comment by mananjadhav** [comment]
Sure I can pick this first thing tomorrow. Quick question about the comment [here](https://github.com/Expensify/App/issues/12893#issuecomment-1490144123) ? Are we good to go ahead with the change?

**Comment by ArekChr** [comment]
> Sure I can pick this first thing tomorrow. Quick question about the comment [here](https://github.com/Expensify/App/issues/12893#issuecomment-1490144123) ? Are we good to go ahead with the change?
> 
> @ArekChr Could you please resolve the conflicts, and I can start testing this?

Hey @mananjadhav, this comment relates to mWeb Safari and Chrome there was an issue "can not zoom-in preview in mWeb".
The PR and issue relate only to Android, and here are no blockers.

**Review by mananjadhav — APPROVED** [review]
Change looks fine. Testing on Android first now.

**Review by cristipaval — APPROVED** [review]

**Comment by mananjadhav** [comment]
I am on the testing @cristipaval

**Comment by mananjadhav** [comment]
I did some basic testing on Android and it seems to be working.

Screenshot 2023-04-12 at 8 08 47 PM

I am going to test this with caching and few other regression tests for images.

**Comment by mananjadhav** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [ ] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>
web-large-image

</details>

<details>
<summary>Mobile Web - Chrome</summary>
mweb-chrome-large-image

</details>

<details>
<summary>Mobile Web - Safari</summary>

mweb-safari-large-image

</details>

<details>
<summary>Desktop</summary>

desktop-large-image

</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>

android-large-image

</details>

**Comment by mananjadhav** [comment]
@francoisl @cristipaval Could you help with the iOS screencasts here please? I am a bit blocked on my setup and I tried a few hours but I couldn't resolve it.

**Comment by francoisl** [comment]
My iOS dev build has also been broken since the latest XCode update, but I can try from Browserstack!

**Comment by mananjadhav** [comment]
@francoisl this is deployed on staging, so I was able to test on my physical device using Testflight. I've uploaded the screencast in the checklist. That should be good right?

**Comment by francoisl** [comment]
Yep thanks. Out of curiosity, did you notice any performance regression? We had [this deploy blocker](https://github.com/Expensify/App/issues/17427) opened by the QA team yesterday, I wonder if it has to do with disabling hardware acceleration.

**Pull request merged**